### PR TITLE
refactoring function

### PR DESCRIPTION
**Pull request by JanaS87**
- adjust the intolerances to suit the use case
- refactored the forbiddenFoodForCat- function to work as expected
- add a second some() to the respective query
- tear the strings into words for a better examination of the transmitted data

### CODE REVIEW

**Review by onemanwenttomow — COMMENTED** [review]
@JanaS87 First of all great effort with this. 🎉
It does not look like an easy task. 

I will be honest in that i can't tell you if the function works as expected or not, as i am not 100% sure of the requirements. (based on what needs to be checked regarding allergies and ingredients). For example. If a cat has an egg allergies, is it ok for them to eat something with `"Eggshell powder (0.5%)"`?

If you are looking to simplify the function at some point, then i think you need to simplify your data structure. 

I.e. each food could have a property like so

``` js
intolerances: ["lactose", // etc]
```

This would make it much easier to check a cats intolerances against the food. 

The same with checking the cat diseases. At the moment your check with splitting twice means you are very reliant on the strings always following the same structure. 

On an additional note. This might be the perfect time to looking into [writing some tests](https://jestjs.io/docs/getting-started) for this function. That way you can give your function some inputs and check them against the expected outputs. 

So in summary. Sorry for not being able to provide any more specifics than this. But fantastic work on getting this far. It must have been a real brain spinner.
